fs/remote/resolver.go

```diff
@@ -450,46 +450,47 @@ func redirect(ctx context.Context, blobURL string, tr http.RoundTripper) (string
 	return "", fmt.Errorf("%w on redirect %v", ErrUnexpectedStatusCode, res.StatusCode)
 }
 
+func DoHeadRequest(ctx context.Context, realURL string, rt http.RoundTripper) (*http.Response, error) {
+	statusCodes := []int{0, 0}
+
+	// Some repos do not allow us to make HEAD calls (e.g.
+	// ECR Public does not allow HEAD calls with default credentials),
+	// so try twice — once with HEAD, once with GET.
+	methods := []string{http.MethodHead, http.MethodGet}
+	for i, method := range methods {
+		req, err := http.NewRequestWithContext(ctx, method, realURL, nil)
+		if err != nil {
+			return nil, err
+		}
+		if method == http.MethodGet {
+			req.Header.Set("Range", "bytes=0-1")
+		}
+
+		resp, err := rt.RoundTrip(req)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
```

Comment:
defer routines are for the scope of the entire function. But it's better to close the http body for each iteration. we can either don't use defer or wrap the logic in a closure.